modin/data_management/data_manager.py

```diff
@@ -1792,15 +1792,21 @@ def rank(self, **kwargs):
         axis = kwargs.get("axis", 0)
         numeric_only = True if axis else kwargs.get("numeric_only", False)
         func = self._prepare_method(pandas.DataFrame.rank, **kwargs)
-        new_data = self.map_across_full_axis(axis, func)
-        # Since we assume no knowledge of internal state, we get the columns
-        # from the internal partitions.
         if numeric_only:
-            new_columns = self.compute_index(1, new_data, True)
+            result, data_manager = self.numeric_function_clean_dataframe(axis)
+            if result is not None:
+                return self.from_pandas(
+                    pandas.DataFrame(index=data_manager.index), type(self.data)
```

Comment:
how is `data_manager.index` chosen here?

Reply:
`data_manager.index` is the same as `self.index`. I changed it to `self.index` for readability